src/extension/src/ActionHandler.py

```diff
@@ -87,10 +87,14 @@ def tear_down(self):
     def setup_telemetry(self):
         # check if events folder exists, if it does init telemetry, if events folder does not exist, log that telemetry is not supported by agent since events folder does not exist
         events_folder = self.ext_env_handler.events_folder
-        if events_folder is None or not os.path.exists(events_folder):
+        if events_folder is None:
             err_msg = Constants.TELEMETRY_AT_AGENT_NOT_COMPATIBLE_ERROR_MSG
             self.logger.log_error(err_msg)
         else:
+            if not os.path.exists(events_folder):
+                os.mkdir(events_folder)
+                self.logger.log("Events folder path found in HandlerManifest but does not exist on disk. Creating now. [Path={0}]".format(str(events_folder)))
```

Comment:
Correction: "Events folder path found in HandlerEnvironment.json but the folder itself does not exist on disk..."

HandlerManifest is a different file containing some other data.

Reply:
Thanks, fixed.